Assets/Scripts/Tank/PlayerTank.cs

```diff
@@ -0,0 +1,98 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerTank : GenericSingleton<PlayerTank>
+{
+    public int m_PlayerNumber = 1;
```

Comment:
We do not need here to make everything public if we need these variables inside the class only.